### PR TITLE
#102 - Add index merging slice

### PR DESCRIPTION
**Pull request by genryxy**
Part of #102 
Add slice for merging two index files

### CODE REVIEW

File: src/main/java/com/artipie/helm/http/IndexMergingSlice.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                    new PublisherAs(this.source).asciiString(),
+                    (rmt, src) -> {
+                        final IndexYamlMapping srcidx = new IndexYamlMapping(src);
+                        for (final String name : rmt.entries().keySet()) {
+                            srcidx.addChartVersions(name, rmt.byChart(name));
+                        }
+                        return srcidx;
+                    }
+                ).thenApply(IndexYamlMapping::toContent)
+                .thenApply(opt -> opt.orElse(Content.EMPTY));
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/helm/http/IndexMergingSlice.java:176]: @genryxy hm, empty content does not seem possible here, is it really necessary to return optional from `IndexYamlMapping.toContent()`?

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy please check my questions

File: src/main/java/com/artipie/helm/http/IndexMergingSlice.java (hunk 1 above)
Comment (genryxy) [reply src/main/java/com/artipie/helm/http/IndexMergingSlice.java:176]: @olenagerasimova it is possible to be empty when we try to generate index by specific directory

File: src/main/java/com/artipie/helm/http/IndexMergingSlice.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        return new AsyncResponse(
+            temp.save(yaml, new Content.From(body))
+                .thenCompose(
+                    ignored -> {
+                        final CompletionStage<Response> res;
+                        if (matcher.matches()) {
+                            final Key location = Key.ROOT;
+                            res = this.storage.exclusively(
+                                location,
+                                target -> this.storage.value(yaml)
```
[the remark this answers is not in the document]
Comment (genryxy) [reply src/main/java/com/artipie/helm/http/IndexMergingSlice.java:97]: @olenagerasimova there is no check that yaml exists. I'll add this one

**Comment by genryxy** [comment]
@olenagerasimova thanks, answered and fixed

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks, looks good to me

**Comment by 0crat** [comment]
Job `gh:artipie/helm-adapter#103` is not assigned, can't get performer

**Pull request merged**